tools/clang/include/clang/SPIRV/SpirvInstruction.h

```diff
@@ -1943,6 +1978,730 @@ class SpirvDemoteToHelperInvocationEXT : public SpirvInstruction {
+
+  uint32_t getDebugFlags() const { return debugFlags; }
+  SpirvDebugType *getReturnType() const { return returnType; }
+  llvm::ArrayRef<SpirvDebugType *> getParamTypes() const { return paramTypes; }
+
+private:
+  // TODO: Replace uint32_t with enum in the SPIRV-Headers once it is available.
+  uint32_t debugFlags;
+  // Return Type is a debug instruction which represents type of return value of
+  // the function. If the function has no return value, this operand is
+  // OpTypeVoid, in which case we will use nullptr for this member.
+  SpirvDebugType *returnType;
+  llvm::SmallVector<SpirvDebugType *, 4> paramTypes;
+};
+
+/// Represents debug information for a template type parameter.
+class SpirvDebugTypeTemplateParameter : public SpirvDebugType {
+public:
+  SpirvDebugTypeTemplateParameter(llvm::StringRef name, SpirvDebugType *type,
+                                  SpirvInstruction *value,
+                                  SpirvDebugSource *source, uint32_t line,
+                                  uint32_t column);
+
+  DEFINE_RELEASE_MEMORY_FOR_CLASS(SpirvDebugTypeTemplateParameter)
+
+  static bool classof(const SpirvInstruction *inst) {
+    return inst->getKind() == IK_DebugTypeTemplateParameter;
+  }
+
+  bool invokeVisitor(Visitor *v) override;
+
+  SpirvDebugType *getActualType() const { return actualType; }
+  void setValue(SpirvInstruction *c) { value = c; }
+  SpirvInstruction *getValue() const { return value; }
+  SpirvDebugSource *getSource() const { return source; }
+  uint32_t getLine() const { return line; }
+  uint32_t getColumn() const { return column; }
+
+private:
+  SpirvDebugType *actualType; //< Type for type param
+  SpirvInstruction *value;    //< Value. It must be null for type.
+  SpirvDebugSource *source;   //< DebugSource containing this type
+  uint32_t line;              //< Line number
+  uint32_t column;            //< Column number
+};
+
+/// Represents debug information for a template type.
+class SpirvDebugTypeTemplate : public SpirvDebugType {
+public:
+  SpirvDebugTypeTemplate(
+      SpirvDebugInstruction *target,
+      const llvm::SmallVector<SpirvDebugTypeTemplateParameter *, 2> &params);
+
+  DEFINE_RELEASE_MEMORY_FOR_CLASS(SpirvDebugTypeTemplate)
+
+  static bool classof(const SpirvInstruction *inst) {
+    return inst->getKind() == IK_DebugTypeTemplate;
+  }
+
+  bool invokeVisitor(Visitor *v) override;
+
+  llvm::SmallVector<SpirvDebugTypeTemplateParameter *, 2> getParams() {
+    return params;
+  }
+  SpirvDebugInstruction *getTarget() const { return target; }
+
+private:
+  // A debug instruction representing class, struct or function which has
+  // template parameter(s).
+  SpirvDebugInstruction *target;
+
+  // Debug instructions representing the template parameters for this
+  // particular instantiation. It must be DebugTypeTemplateParameter
+  // or DebugTypeTemplateTemplateParameter.
+  // TODO: change the type to SpirvDebugType * when we support
```

Comment:
Is this still relevant? If yes, please file an issue so we don't forget about it.

Reply:
done